src/mono/mono/utils/write-protect.c

```diff
@@ -0,0 +1,27 @@
+/**
+ * \file
+ */
+
+#include "config.h"
+
+#include "mono-compiler.h"
+#include "write-protect.h"
+
+#if defined (HOST_MACCAT) && defined (__aarch64__)
+
+/* our own declaration of pthread_jit_write_protect_np so that we don't see the __API_UNAVAILABLE__ header */
```

Comment:
I'm surprised this doesn't conflict with pthread.h, which is presumably also included by this point.